server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -413,11 +413,11 @@ protected void executeNextPhase(SearchPhase currentPhase, Supplier<SearchPhase>
     private void executePhase(SearchPhase phase) {
         try {
             phase.run();
-        } catch (Exception e) {
+        } catch (RuntimeException e) {
```

Comment:
what's the motivation for this change? is it that `run` no longer throws `IOException`?

Reply:
Yea just to make it clear in code here that we only have `RuntimeException`. Ideally this should never throw because we have a listener injected in the phase that it itself resolves, so while I can't guarantee that it never will, at least the compiler now guarantees that there won't be any checked exceptions :)